Python/ibmcloudsql/SQLQuery.py

```diff
@@ -136,72 +211,281 @@ def _send_req(self, json_data):
+                    }
+                ]
+            }
+            \"\"\"
+            # error code information: https://cloud.ibm.com/apidocs/sql-query
+
+
+        """
         self.logon()
+        sql_text = sql_stmt
         sqlData = {'statement': sql_text}
+
+        def INTO_is_present(sql_text):
+            """ check if INTO keyword is present in the SQL query"""
+            return (" INTO " in sql_text.upper()) or (
+                "\nINTO " in sql_text.upper())
+
         # If a valid pagesize is specified we need to append the proper PARTITIONED EVERY <num> ROWS clause
-        if pagesize or pagesize==0:
-            if type(pagesize) == int and pagesize>0:
-                if self.target_cos:
-                    sqlData["statement"] += " INTO {}".format(self.target_cos)
-                elif " INTO "  not in sql_text.upper():
-                    raise SyntaxError("Neither resultset_target parameter nor \"INTO\" clause specified.")
+        if pagesize or pagesize == 0:
+            if type(pagesize) == int and pagesize > 0:
+                if self.target_cos_url and not INTO_is_present(sql_text):
+                    sqlData["statement"] += " INTO {}".format(self.target_cos_url)
+                elif not INTO_is_present(sql_text):
+                    raise SyntaxError(
+                        "Neither resultset_target parameter nor \"INTO\" clause specified."
+                    )
                 elif " PARTITIONED " in sql_text.upper():
-                    raise SyntaxError("Must not use PARTITIONED clause when specifying pagesize parameter.")
-                sqlData["statement"] += " PARTITIONED EVERY {} ROWS".format(pagesize)
+                    raise SyntaxError(
+                        "Must not use PARTITIONED clause when specifying pagesize parameter."
+                    )
+                sqlData["statement"] += " PARTITIONED EVERY {} ROWS".format(
+                    pagesize)
             else:
-                raise ValueError('pagesize parameter ({}) is not valid.'.format(pagesize))
-        elif self.target_cos:
-            sqlData.update({'resultset_target': self.target_cos})
+                raise ValueError(
+                    'pagesize parameter ({}) is not valid.'.format(pagesize))
+        elif self.target_cos_url and not INTO_is_present(sql_text):
+            sqlData.update({'resultset_target': self.target_cos_url})
 
+        max_tries = self.max_tries
         intrumented_send = backoff.on_exception(
             backoff.expo,
             RateLimitedException,
-            max_tries=self.max_tries
+            max_tries=max_tries
         )(self._send_req)
-
         return intrumented_send(sqlData)
 
+    @check_saved_jobs_decorator
+    def submit_sql_with_checking_saved_jobs(self, sql_stmt, pagesize=None, prefix=None):
```

Comment:
Method name is and signature is not revealing the semantics sufficiently. It only works when project lib is used. Also not sure whether documentation is clear. What is the "The content of previous queries" that is being "stored in a file"?  Is that just the job_id of previous queries? Make that clear. Parameter "prefix" is way too generic. Make clear it refers to a name of previous query that was stored via project lib.

Reply:
Done in commit `4c3eb6593e7ab5a1d2f9f0cfe3ac084cc1ac0b68`.

I revised the docstring to explain the API, also I renamed to `submit_and_track_sql`, and `prefix` option is renamed to `file_name`.